meetings/2018-01-17/minutes.md

```diff
@@ -0,0 +1,417 @@
+features. It is purely focused on IDE support (completion, reverse lookup,
+etc), and does not manage building features.
+
+It was funded by the sponsorship program which also supported a shared LSP
+client for VSCode/Atom, though this funding has basically run out. Ensime has
+support for exporting build info from sbt/Gradle/Maven.
+
+The indexer is the key component for many of the LSP semantic calls, and this
+should move to using SemanticDB as an interchange format. To provide all of the
+IDE features you need more than just the protocol layer, much of which would be
+shared.
+
+Rory added, during the call, that Ensime creates an index, much like the
+SemanticDB index, but predating SemanticDB. Ensime could support all clients
+once LSP support is available.
+
+### IntelliJ IDE - Nikolay Tropin, JetBrains
+
+Nikolay is the team leader for the Scala plugin for IntelliJ, which provides
+support for Scala in IntelliJ. The IntelliJ plugin does not depend on Scalac,
+instead using its own implementation of the Scala typechecker for providing
+presentation support in the IntelliJ editor. IntelliJ also offers Java support.
+
+JetBrains are very interested in LSP and BSP, though they're not sure whether
+they can use it in IntelliJ, which currently relies on their own
+implementation. He noted that Bloop may be particularly useful for IntelliJ,
+though is uncertain about whether it could easily replace IntelliJ's own
+typechecker.
+
+### ScalaIDE/Eclipse - Iulian Dragos, Triplequotes
+
+Iulian has been working on the LSP implementation for VSCode, based on Ensime.
+The implementation delegates the protocol as an in-process Ensime instance,
+circumventing the socket protocol.
+
+This approach avoided many of the problems Iulian faced working on the ScalaIDE
+for Eclipse (which he spent approximately four years on). The ScalaIDE used the
+Scalac presentation compiler directly, through an asynchronous API that was
+able to interact directly with the typechecker. It worked happily with
+incomplete on incorrect programs, providing errors as the user typed, and
+offering early feedback.
+
+All ASTs were available to the IDE, and offered advanced, precise and rich
+information. Unfortunately there was a tight coupling between the plugin and
+the compiler. Memory requirements were high, because each different classpath
+required a different instance of the compiler, without sharing of (for example)
+symbol tables. Iulian is very much in favor of the language-server approach as
+it avoids many of these problems, though having a presentation compiler also
+provides many benefits, particularly for providing feedback for incomplete or
+non-compiling programs.
+
+### Dotty IDE/VSCode - Guillaume Martres, EPFL
+
+Any Dotty project built with sbt can launch VSCode with a single command, and
+will automatically provide common IDE features like code-linking and
+completion. Guillaume suggests that he would prefer the IDE to interact
```

Comment:
Can IDE be replaced with "editor" here? If I read "IDE to interact directly with the language server" correctly then language server refers to Dotty IDE